packages/cubejs-api-gateway/src/gateway.ts

```diff
@@ -1058,17 +1058,25 @@ class ApiGateway {
           'A user\'s query must contain at least one query param.'
         );
       }
-
+      
       query = {
         ...query,
         requestId: context.requestId
       };
 
+      if (!query.limit || !Number.isInteger(query.limit) || query.limit > 50000 || query.limit < 1) {
+        throw new UserError(
+          'A user\'s query must contain limit query param and it must be positive number less than 50000.'
+        );
+      }
+
       if (query.resultFilter?.objectTypes && !Array.isArray(query.resultFilter.objectTypes)) {
         throw new UserError(
           'A query.resultFilter.objectTypes must be an array of strings'
         );
       }
+
+      query.query = `${query.query.replace(';', '')} LIMIT ${query.limit};`;
```

Comment:
Maybe it's better to do `SELECT * FROM (${query.query}) LIMIT ${query.limit}` instead of hacking?)

Reply:
yeah we think about it, but will it work on all db types?